### PR TITLE
A user can see the creator of the list

### PR DESCRIPTION
**Pull request by anatoliikryvishyn**

### CODE REVIEW

**Comment by anatoliikryvishyn** [comment]
@serhiir, please review and test [here](https://tada-interlink.herokuapp.com/).

**Comment by serhiir** [comment]
@anatoliikryvishyn Please make `grey` color and increase the indednt above, like the following:
adjustment

**Comment by serhiir** [comment]
@anatoliikryvishyn, a pattern was:
> :clock1: Created by *tada@tada.com* on Jan 18, 2016 

So please use short name of the month.

File: config/locales/en.yml [hunk 1]
```diff
@@ -19,4 +19,7 @@ en:
     modal_description_label: "Description"
     create_btn: "Create list"
     create_message: "Create the list and start adding a new ToDo items"
-    no_list_notification: "No Todo lists were created yet, create your first one"
\ No newline at end of file
+    no_list_notification: "No Todo lists were created yet, create your first one"
+    creation_information: "Created by <span class='creator-email'>%{username}</span> on %{date}"
```
Comment (serhiir) [inline config/locales/en.yml]: Nnoooo! :) @anatoliikryvishyn Texting should be independent from the layout. No html here! Please put the layout in the view and texts keep clean here.

File: app/controllers/lists_controller.rb [hunk 2]
```diff
@@ -1,11 +1,13 @@
 class ListsController < ApplicationController
   def index
-    lists = List.ordered_by_creation.all_with_items
+    user = Users::CurrentUser.new (current_user)
+    lists = user.lists.ordered_by_creation.all_with_items
     @lists = lists.map { |list| ListView.new(list) }
```
Comment (serhiir) [inline app/controllers/lists_controller.rb]: @anatoliikryvishyn, I think there is no such requirement to change given behavior (at least within this PR). I suggest to revert the change and simply leave as it was before.

File: app/controllers/lists_controller.rb [hunk 3]
```diff
@@ -1,11 +1,13 @@
...
   end
 
   def create
-    list = List.new(list_params)
+    user = Users::CurrentUser.new (current_user)
+    list = user.createList list_params
```
Comment (serhiir) [inline app/controllers/lists_controller.rb]: @anatoliikryvishyn As you use simple approach with single user, I'd suggest to do not introduce `Users::CurrentUser` entity at all. All you need here for now is simply set `user_id` using `current_user` (via `list_params` or directly).

File: app/logic/users/current_user.rb [hunk 4]
```diff
@@ -0,0 +1,25 @@
+module Users
+  class CurrentUser
+    def initialize user
+      @user = user
+    end
```
Comment (serhiir) [inline app/logic/users/current_user.rb]: @anatoliikryvishyn As it was mentioned, I'd get rid of given class.

File: app/models/list.rb [hunk 5]
```diff
@@ -4,6 +4,7 @@ class List < ActiveRecord::Base
   validates_presence_of :title
 
   scope :ordered_by_creation, -> { order(created_at: :desc) }
+  scope :of_user, -> (user_id) { where(user_id: user_id)}
```
Comment (serhiir) [inline app/models/list.rb]: @anatoliikryvishyn :+1:

File: app/models/users/user.rb [hunk 6]
```diff
@@ -0,0 +1,11 @@
+module Users
+  class User
+    attr_reader :id, :email, :password
+
+    def initialize (options = {})
+      @id = options.fetch(:id)
+      @email = options.fetch(:email)
+      @password = options.fetch(:password)
```
Comment (serhiir) [inline app/models/users/user.rb]: @anatoliikryvishyn Don't you think that the following would serve http://api.rubyonrails.org/classes/ActiveRecord/Base.html#class-ActiveRecord::Base-label-Creation?
So it seems that such ability is already built in.

Also, the main case of `fetch` using, I believe, is using of default value.

File: app/view_objects/list_view.rb [hunk 7]
```diff
@@ -30,6 +30,15 @@ def item_creation_text
...
   end
 
+  def created_by
+    user.email
+  end
+
+  def created_at
+    date = list.created_at
+    date.strftime I18n.t 'lists.created_at.format'
+  end
```
Comment (serhiir) [inline app/view_objects/list_view.rb]: @anatoliikryvishyn :+1: Good to see a new code to displaying purpose here.

Comment (serhiir) [inline app/view_objects/list_view.rb]: @anatoliikryvishyn I think `_at` points to time (datetime), but here is a date only. I'd rename into `created_on`.

File: app/view_objects/list_view.rb [hunk 8]
```diff
@@ -37,4 +46,9 @@ def item_creation_text
   def items
     list.items
   end
+
+  def user
+    user_id = list.user_id
+    Users::User.new({ id: user_id, email:'tada@tada.com', password: '123456789' })
+  end
```
Comment (serhiir) [inline app/view_objects/list_view.rb]: @anatoliikryvishyn Just FYI: Make sure that in future integration with a real user object on board here will be no queries into DB.

File: app/models/list.rb (hunk 5 above)
Comment (serhiir) [inline app/models/list.rb]: Note, that if we remove `Users::CurrentUser` class , this scope will have no usages.

File: db/migrate/20160121142136_add_user_to_list.rb [hunk 9]
```diff
@@ -0,0 +1,9 @@
+class AddUserToList < ActiveRecord::Migration
+  def change
+    add_column :lists, :user_id, :integer, default: 1
+  end
```
Comment (serhiir) [inline db/migrate/20160121142136_add_user_to_list.rb]: @anatoliikryvishyn I'd not recommend such default value as it might cause problems in the future.
I think you even do not need given migration as a user is faked.

File: test/view_objects/list_view_test.rb [hunk 10]
```diff
@@ -24,4 +24,15 @@ class ListViewTest < ActiveSupport::TestCase
...
+  test "creator of list" do
+    list = ListView.new (build(:home))
+
+    assert_equal 'tada@tada.com' ,list.created_by
+  end
+
+  test "list's creation date" do
+    list = ListView.new (build(:home))
+
+    assert_equal "April 12, 2015" ,list.created_at
```
Comment (serhiir) [inline test/view_objects/list_view_test.rb]: @anatoliikryvishyn Instantiation of `list` is duplicating, please DRY (move into setup).

Comment (serhiir) [inline test/view_objects/list_view_test.rb]: @anatoliikryvishyn Please format asserts so they have proper indentations.

File: config/locales/en.yml (hunk 1 above)
Comment (serhiir) [inline config/locales/en.yml]: @anatoliikryvishyn To clarify this comment: you can simply compose needed message from a few keys or use just palce holders in the message without html.

**Comment by serhiir** [comment]
@anatoliikryvishyn Verified and reviewed.

File: app/models/users/user.rb (hunk 6 above)
Comment (anatoliikryvishyn) [inline app/models/users/user.rb]: @serhiir, as I can see, and experimented, this could work if our class was inherited from ``ActiveRecord::Base``, but as it is fake class, we don`t need it to be in data base.

**Comment by anatoliikryvishyn** [comment]
@serhiir, updated. [Link](https://tada-interlink.herokuapp.com/) on server to test.

**Comment by serhiir** [comment]
@anatoliikryvishyn I suppose [the app](https://tada-interlink.herokuapp.com/) is updated after last code update, right?

**Comment by anatoliikryvishyn** [comment]
>@anatoliikryvishyn I suppose the app is updated after last code update, right?

@serhiir, yes, it is.
